File: android/ui-framework-sample/ui-framework/src/main/java/kh/mcnc/lib/components/BizRibbonButton.kt
package kh.mcnc.lib.components

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

/**
 * BizRibbonButton - Jetpack Compose component matching Web UI-Framework `PPRibbonButton.vue`.
 */
@Composable
fun BizRibbonButton(
    modifier: Modifier = Modifier,
    label: String = "",
    icon: String = "",
    iconSvg: String = "",
    size: String = "large",
    active: Boolean = false,
    disabled: Boolean = false,
    onClick: () -> Unit = {},
    content: @Composable (() -> Unit)? = null
) {
    val primaryNavy = Color(0xFF1A2A5E)
    val primaryBlue = Color(0xFF003399)
    val secondaryBg = Color(0xFFF1F5F9)
    val borderColor = Color(0xFFCBD5E1)

    Box(
        modifier = modifier
            .fillMaxWidth()
            .defaultMinSize(minHeight = 44.dp)
            .background(Color.White, shape = RoundedCornerShape(12.dp))
            .border(1.dp, borderColor, shape = RoundedCornerShape(12.dp))
            .padding(16.dp)
            .clickable(onClick = onClick),
        contentAlignment = Alignment.CenterStart
    ) {
        Row(
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.SpaceBetween,
            modifier = Modifier.fillMaxWidth()
        ) {
            Column(modifier = Modifier.weight(1f)) {
                Text(
                    text = "RibbonButton",
                    fontSize = 16.sp,
                    fontWeight = FontWeight.SemiBold,
                    color = primaryNavy
                )
                if (content != null) {
                    Spacer(modifier = Modifier.height(8.dp))
                    content()
                }
            }
        }
    }
}

/**
 * Web UI-Framework parity alias for [BizRibbonButton].
 */
@Composable
fun PPRibbonButton(
    modifier: Modifier = Modifier,
    label: String = "",
    icon: String = "",
    iconSvg: String = "",
    size: String = "large",
    active: Boolean = false,
    disabled: Boolean = false,
    onClick: () -> Unit = {},
    content: @Composable (() -> Unit)? = null
) = BizRibbonButton(
        modifier = modifier,
        label = label,
        icon = icon,
        iconSvg = iconSvg,
        size = size,
        active = active,
        disabled = disabled,
        onClick = onClick,
        content = content
)

File: ios/ui-framework-sample/UIFramework/Sources/UIFramework/BizRibbonButton.swift
import SwiftUI

/**
 * BizRibbonButton - SwiftUI component matching Web UI-Framework `PPRibbonButton.vue`.
 */
public struct BizRibbonButton: View {
    public var label: String
    public var icon: String
    public var iconSvg: String
    public var size: String
    public var active: Bool
    public var disabled: Bool
    public var action: () -> Void

    public init(
        label: String = "",
        icon: String = "",
        iconSvg: String = "",
        size: String = "large",
        active: Bool = false,
        disabled: Bool = false,
        action: @escaping () -> Void = {}
    ) {
        self.label = label
        self.icon = icon
        self.iconSvg = iconSvg
        self.size = size
        self.active = active
        self.disabled = disabled
        self.action = action
    }

    public var body: some View {
        Button(action: action) {
            VStack(alignment: .leading, spacing: 8) {
                Text("RibbonButton")
                    .font(.system(size: 16, weight: .semibold))
                    .foregroundColor(Color(hex: "#1A2A5E"))
            }
            .frame(maxWidth: .infinity, alignment: .leading)
            .padding(16)
            .background(Color.white)
            .cornerRadius(12)
            .overlay(
                RoundedRectangle(cornerRadius: 12)
                    .stroke(Color(hex: "#CBD5E1"), lineWidth: 1)
            )
        }
        .buttonStyle(PlainButtonStyle())
    }
}

/**
 * Web UI-Framework parity alias for [BizRibbonButton].
 */
public typealias PPRibbonButton = BizRibbonButton
